scvi/model/_destvi.py

```diff
@@ -0,0 +1,282 @@
+import logging
+from typing import Optional, OrderedDict, Union
+
+import numpy as np
+import pandas as pd
+import torch
+from anndata import AnnData
+
+from scvi.data import register_tensor_from_anndata
+from scvi.model import CondSCVI
+from scvi.model.base import BaseModelClass, UnsupervisedTrainingMixin
+from scvi.module import MRDeconv
+
+logger = logging.getLogger(__name__)
+
+
+class DestVI(UnsupervisedTrainingMixin, BaseModelClass):
+    """
+    Multi-resolution deconvolution of Spatial Transcriptomics data (DestVI).
+
+    Parameters
+    ----------
+    st_adata
+        spatial transcriptomics AnnData object that has been registered via :func:`~scvi.data.setup_anndata`.
+    cell_type_mapping
+        mapping between numerals and cell type labels
+    decoder_state_dict
+        state_dict from the decoder of the CondSCVI model
+    px_decoder_state_dict
+        state_dict from the px_decoder of the CondSCVI model
+    px_r
+        parameters for the px_r tensor in the CondSCVI model
+    n_hidden
+        Number of nodes per hidden layer.
+    n_latent
+        Dimensionality of the latent space.
+    n_layers
+        Number of hidden layers used for encoder and decoder NNs.
+    **module_kwargs
+        Keyword args for :class:`~scvi.modules.MRDeconv`
+
+    Examples
+    --------
+    >>> sc_adata = anndata.read_h5ad(path_to_scRNA_anndata)
+    >>> scvi.data.setup_anndata(sc_adata)
+    >>> sc_model = scvi.CondSCVI(sc_adata)
+    >>> mean_vprior, var_vprior = sc_model.get_vamp_prior(sc_adata, p=100)
+    >>> st_adata = anndata.read_h5ad(path_to_ST_anndata)
+    >>> scvi.data.setup_anndata(st_adata)
+    >>> spatial_model = DestVI.from_rna_model(st_adata, sc_model, mean_vprior=mean_vprior, var_vprior=var_vprior)
```

Comment:
couldn't the vamp prior be an option of from rna model to reduce that vamp prior liine of code?

Reply:
yes good idea